person.rb

```diff
@@ -0,0 +1,37 @@
+# Class containing data about a person
+require_relative 'nameable'
+require './rental'
+
+class Person < nameable
+  # Fixed the class name to inherit from Nameable (capitalized)
+  attr_accessor :name, :age
+  attr_reader :id
+
+  def initialize(age, name = 'Unknown', parent_permission: true)
+    super() # Call the superclass constructor with no arguments
+    @id = rand(1..1000)
+    @name = name
+    @age = age
+    @parent_permission = parent_permission
+  end
+
+  def add_rental(rental)
```

Comment:
- [ ] Hey there! 😊 Remember that fantastic many-to-many relationship between Person and Book? 📚👥 To achieve the has-many association with Rentals in the `class Person`, it's crucial to ensure that the method for adding rentals accepts the person itself as a parameter when creating the new rental. This way, you'll establish a solid link between `Person` and its many `Rentals`, making your project's associations crystal clear! If you have any questions or need assistance with the implementation, don't hesitate to ask. Happy coding! 🚀📖
```suggestion
    def add_rental(book, date)
    # *The person itself should be sent as a parameter to create the new rental achieving the 'has-many' association
    Rental.new(date, book, self)
```

Reply:
changes made accordingly